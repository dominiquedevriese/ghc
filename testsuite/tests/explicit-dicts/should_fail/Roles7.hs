{-# LANGUAGE TypeApplications #-}
{-# LANGUAGE TypeFamilies #-}
module Roles7 where

import GHC.Base (Ord(..))

reverseOrd :: Ord a => Ord.Dict a
reverseOrd = undefined

type family F a

f7 = (undefined :: Ord a => a -> F a -> F a)
     ((reverseOrd @Int :: Ord a))
