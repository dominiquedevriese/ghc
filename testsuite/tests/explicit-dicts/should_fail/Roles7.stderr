
Roles7.hs:12:6: error:
    • Explicit dictionary application to: ‘Ord a’
      is not allowed because of instance incoherence:
      In the function type ‘a -> F a -> F a’
        Type variable ‘a’ has role Nominal
    • In the expression:
        (undefined :: Ord a => a -> F a -> F a)
          ((reverseOrd @Int :: Ord a))
      In an equation for ‘f7’:
          f7
            = (undefined :: Ord a => a -> F a -> F a)
                ((reverseOrd @Int :: Ord a))
